cpp/src/join/nested_loop_join.cuh

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/detail/utilities/cuda.cuh>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/table/table_view.hpp>
+#include <iostream>
+
+#include <join/hash_join.cuh>
+#include <join/join_common_utils.hpp>
+#include <join/join_kernels.cuh>
+
+namespace cudf {
+namespace detail {
+/* --------------------------------------------------------------------------*/
+/**
+ * @brief  Gives an estimate of the size of the join output produced when
+ * joining two tables together.
+ *
+ * @throw cudf::logic_error if JoinKind is not INNER_JOIN or LEFT_JOIN
+ *
+ * @param left The left hand table
+ * @param right The right hand table
+ *
```

Comment:
```suggestion
 * @tparam join_kind The type of join to be performed
 *
```

Reply:
I committed this locally. In my understanding, it should be `JoinKind` (the name of the template parameter), not `join_kind` (the type of the template parameter).